operator/main.go

```diff
@@ -60,23 +60,40 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+func setupLogger(debug bool) {
+	// NOTE: The Go logger doesn't use `DEBUG`, `WARNING`, etc.  but we can mimic
+	// it to maintain compatibility.
+	logLevel := os.Getenv("SELDON_LOG_LEVEL")
+
+	if logLevel == "DEBUG" || logLevel == "INFO" {
+		debug = true
+	} else if logLevel == "WARN" || logLevel == "WARNING" || logLevel == "ERROR" {
+		debug = false
+	}
+
+	ctrl.SetLogger(zap.Logger(debug))
+}
+
 func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var webHookPort int
 	var namespace string
 	var operatorNamespace string
 	var createResources bool
+	var debug bool
+
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.IntVar(&webHookPort, "webhook-port", 443, "Webhook server port")
 	flag.StringVar(&namespace, "namespace", "", "The namespace to restrict the operator.")
 	flag.StringVar(&operatorNamespace, "operator-namespace", "default", "The namespace of the running operator")
 	flag.BoolVar(&createResources, "create-resources", false, "Create resources such as webhooks and configmaps on startup")
+	flag.BoolVar(&debug, "debug", false, "Enable debug mode. Logs will be more verbose, and not structured.")
```

Comment:
Reaching to this point I am a bit confused expected behaviour. 
Do I understand correctly that there are basically two modes:
- debug = true
- debug = false

that just can be overwritten by environmental variable `SELDON_LOG_LEVEL` such that:
- `DEBUG`, `INFO` -> debug = true
- `WARN`, `WARNING` -> debug = false
?

Reply:
Yeah, that was my initial idea so that the high-level interface is the same for the user. That is, they can set a `SELDON_LOG_LEVEL` on any component. 

However, as I mentioned above it may make sense to decouple debug and the log level.